### PR TITLE
fix: root affected level

### PR DESCRIPTION
**Pull request by mkundu1**
Distinguishing between `initialize` and `reinitialize`:
`initialize` - A different workflow is created, root affected callback is cleared.
`reinitialize` - The same workflow is reinitialized, root affected callback is not cleared.

### CODE REVIEW

File: src/ansys/fluent/core/session_pure_meshing.py [hunk 1]
```diff
@@ -117,17 +117,17 @@ def workflow(self):
 
     def watertight(self):
         """Get a new watertight workflow."""
-        self._base_meshing.watertight_workflow.reinitialize()
+        self._base_meshing.watertight_workflow.initialize()
```
Comment (mkundu1) [inline src/ansys/fluent/core/session_pure_meshing.py:120]: Just curious, is there any benefit in returning the previous instance of the `WatertightMeshingWorkflow` class? If `watertight()` always creates a new instance of `WatertightMeshingWorkflow`, we could clear the root-affected callback at the `WatertightMeshingWorkflow.__init__` method.

Comment (seanpearsonuk) [reply src/ansys/fluent/core/session_pure_meshing.py:120]: Yes, the current code seems unnecessarily complicated. Why is it telling objects to initialize and reinitialize etc?

Comment (prmukherj) [reply src/ansys/fluent/core/session_pure_meshing.py:120]: We can discuss this once in today's meeting.

**Review by hpohekar — APPROVED** [review]

File: src/ansys/fluent/core/session_pure_meshing.py (hunk 1 above)
Comment (prmukherj) [reply src/ansys/fluent/core/session_pure_meshing.py:120]: @mkundu1, Please put in a github issue for refactoring this. Thank you

**Review by prmukherj — APPROVED** [review]

**Pull request merged**